abap-code-review/src/code_review.adoc

```diff
@@ -0,0 +1,62 @@
+
+=== Why perform Code Reviews?
+
+Code reviews can detect issues in the software that are harder to find via other methods:
+
+* Foster link:https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md[Clean code] to improve *maintainability* and *readability* of code, including comments, and to avoid unnecessary *complexity*.
+* Ensure the code is *well designed*.
+* Consider behavioral aspects like *performance* and *security*.
+* Improve *testability* of code, including test design and adequateness of tests.
+
+Code reviews are one possibility to *share experience and knowledge* between developers. It works a bit like "asynchronous pair programming", so that colleagues can learn from each other, and share know-how about all parts of the code.
+
+Code reviews help to prevent that *code health* deteriorates over time.
+
+=== Size and Speed of Reviews
+
+Change-based code reviews are these days considered more effective than formal link:https://en.wikipedia.org/wiki/Fagan_inspection[Fagan-style code inspections] as they were practiced some years ago. Conducted continuously, they require less effort while still finding a considerable number of defects at an earlier time.The main reason is the *smaller size* of code deltas to digest. This ensures that reviews can be performed both, more *quickly* and *thoroughly*, while keeping the process lean, i.e., less waste if changes are rejected and lower probability to block other development. Changes should be self-contained to help reviewers focus on the task at hand.
+
+An *ABAP Transport Request* typically fulfills these criteria.
+
+Code reviews need to be done *fast*. Hence, you need to balance between the interruption of your own work against the probability of blocking the release of other developers' transport. Rule of thumb for the maximum delay should be *one day*. Usually, the speed of code reviews will improve over time.
+
+=== How to Select Reviewers?
+
+In general, reviewers can be selected manually or automatically. Manual selection happens either by the author explicitly requesting reviews from colleagues with matching expertise, or by interested reviewers actively picking open changes for review. The latter requires a certain "meritocracy" culture in an organization, where novice developers may gain reputation by reviewing open changes.
+
+*Automated selection* of code reviewers is also possible, for instance, via round robin routing or advanced concepts like link:https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners[Code Owners]. The latter enables you to define (groups of) default reviewers for certain files and folders on git level. This can be useful for critical coding parts or known hot spots.
```

Comment:
Should we also add code-owners for the code review section?

I think it will also become relevant when we add Clean UI5.

Reply:
as infrastructure for the "styleguide" repository? Yea, I think so, but suggest moving it to a different PR/issue

Reply:
"Manual selection happens either by the author (...), or by interested reviewers"
 I have worked in an organization, where the team leader who assigned the task to a specific developer, also decided who is doing the code review.

Reply:
> "Manual selection happens either by the author (...), or by interested reviewers"
> I have worked in an organization, where the team leader who assigned the task to a specific developer, also decided who is doing the code review.

Sounds not very agile to me, but, yes, this could be covered as well

Reply:
> Should we also add code-owners for the code review section?
> 
> I think it will also become relevant when we add Clean UI5.

We can add this later, once the initial PR is merged

Reply:
done with https://github.com/xtough/styleguides/commit/72e265847b90c07f51da3060796424ac11c84bf8